src/colmap/estimators/bundle_adjustment.cc

```diff
@@ -839,6 +842,211 @@ void RigBundleAdjuster::ParameterizeCameraRigs(Reconstruction* reconstruction) {
   }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// PosePriorBundleAdjuster
+////////////////////////////////////////////////////////////////////////////////
+
+PosePriorBundleAdjuster::PosePriorBundleAdjuster(
+    const BundleAdjustmentOptions& options,
+    const Options& prior_options,
+    const BundleAdjustmentConfig& config,
+    const std::unordered_map<image_t, PosePrior>& image_id_to_pose_prior)
+    : BundleAdjuster(options, config),
+      prior_options_(prior_options),
+      image_id_to_pose_prior_(image_id_to_pose_prior) {
+  setRansacMaxErrorFromPriorsCovariance();
+}
+
+bool PosePriorBundleAdjuster::Solve(Reconstruction* reconstruction) {
+  loss_function_ =
+      std::unique_ptr<ceres::LossFunction>(options_.CreateLossFunction());
+
+  if (prior_options_.use_robust_loss_on_prior_position) {
+    prior_loss_function_ = std::make_unique<ceres::CauchyLoss>(
+        prior_options_.prior_position_loss_scale);
+  }
+
+  // Initialize images' position w.r.t. priors with a rigid sim3D alignment
+  Sim3DAlignment(reconstruction);
```

Comment:
I suggest making `Sim3DAlignment` return the success:
```suggestion
  use_prior_position_ = Sim3DAlignment(reconstruction);
```
This is cleaner in case `Solve` is called multiple times (currently it relies on default values of `use_prior_position_`).

Reply:
:+1: